driver-core/src/test/unit/com/mongodb/internal/connection/ServerDiscoveryAndMonitoringMonitoringTest.java

```diff
@@ -70,6 +70,7 @@ public void shouldPassAllOutcomes() {
             }
             BsonDocument outcome = phase.asDocument().getDocument("outcome");
             assertEvents(outcome.getArray("events"));
+            clusterListener.clearClusterDescriptionChangedEvents();
```

Comment:
I think this should be moved to afterTest / beforeTest (setUp/cleanUp), so that its always run even if the assertEvents method fails.

Reply:
The changed events needs to be cleared after each phase, so I have added a `try-finally` block to ensure this.

Reply:
:+1: